sdk/python/generative-ai/rag/notebooks/custom_crack_and_chunk/crack_and_chunk_with_doc_intel/crack_and_chunk.py

```diff
@@ -0,0 +1,57 @@
+import os
+import traceback
+
+from azureml.rag.documents.cracking import file_extension_loaders
+from azureml.rag.tasks.crack_and_chunk import __main__, crack_and_chunk_arg_parser, str2bool
+from azureml.rag.tasks.crack_and_chunk import main as main_crack_and_chunk
+from azureml.rag.utils.connections import get_connection_by_id_v2
+from azureml.rag.utils.logging import (
+    get_logger,
+    safe_mlflow_start_run,
+    track_activity,
+)
+
+from azure.ai.formrecognizer import DocumentAnalysisClient
+from azure.core.credentials import AzureKeyCredential
+from document_intelligence_loader import DocumentIntelligencePDFLoader
+
+logger = get_logger("crack_and_chunk_document_intelligence")
+
+
+def main(args, logger, activity_logger):
+    if args.doc_intel_connection_id:
+        document_intelligence_connection = get_connection_by_id_v2(args.doc_intel_connection_id)
+
+        os.environ["DOCUMENT_INTELLIGENCE_ENDPOINT"] = document_intelligence_connection["properties"]["metadata"]["endpoint"]
+        os.environ["DOCUMENT_INTELLIGENCE_KEY"] = document_intelligence_connection["properties"]["credentials"]["keys"]["api_key"]
+        os.environ["AZURE_AI_DOCUMENT_INTELLIGENCE_USE_LAYOUT"] = str(args.use_layout)
+
+        DocumentIntelligencePDFLoader.document_intelligence_client = DocumentAnalysisClient(
+            endpoint=document_intelligence_connection["properties"]["metadata"]["endpoint"],
+            credential=AzureKeyCredential(document_intelligence_connection["properties"]["credentials"]["keys"]["api_key"])
+        )
+        DocumentIntelligencePDFLoader.use_layout = args.use_layout
+    else:
+        raise ValueError("doc_intel_connection_id is required")
+
+    # Override default `.pdf` loader to use Azure AI Document Intelligence
+    file_extension_loaders[".pdf"] = DocumentIntelligencePDFLoader
```

Comment:
New to Python - but this seems like all sorts of bad code smells.
I guess versions are pinned so it's okay though?

Reply:
This is the intent of the `file_extension_loaders` map, so that more formats can added and existing ones can be modified.

It would be possible to clone the set and pass it into DocumentChunksIterator manually, but I was trying to keep this sort of modification simple.